app/src/androidTest/java/com/eis/audioUtility/AudioUtilityManagerTest.java

```diff
@@ -0,0 +1,167 @@
+package com.eis.audioUtility;
+
+
+import android.content.Context;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Francesco Bau', Alberto Ursino
+ */
+public class AudioUtilityManagerTest {
+
+    Context context;
+    private static final int INVALID_PERCENTAGE_LOW = -1;
+    private static final int INVALID_PERCENTAGE_HIGH = 101;
+
+    @Before
+    public void init() {
+        context = InstrumentationRegistry.getInstrumentation().getTargetContext();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void setAlarmVolume_percentage_isTooLow() {
+        AudioUtilityManager.setVolume(context, AudioUtilityManager.AUMStream.ALARM, INVALID_PERCENTAGE_LOW);
+        Assert.fail("Should have thrown IllegalArgumentException.");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void setAlarmVolume_percentage_isTooHigh() {
+        AudioUtilityManager.setVolume(context, AudioUtilityManager.AUMStream.ALARM, INVALID_PERCENTAGE_HIGH);
+        Assert.fail("Should have thrown IllegalArgumentException");
+    }
+
+    @Test
+    public void setAlarmVolumeToMax_maxVolume_isOK() {
+        try {
+            AudioUtilityManager.setVolumeToMax(context, AudioUtilityManager.AUMStream.ALARM);
+            //Success
+        } catch (IllegalArgumentException e) {
+            Assert.fail(e.getStackTrace() + "\nMAX_PERCENTAGE is NOT OK. Please check this value.");
+        } catch (Exception x) {
+            Assert.fail(x.getStackTrace() + "\nIt's NOT supposed to fail like this!");
+        }
```

Comment:
Also here, any exception is automatically a test failed, you don't need to block it and make the test fail yourself

Reply:
Thank you again.